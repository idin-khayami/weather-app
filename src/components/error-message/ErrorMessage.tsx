import clsx from 'clsx';

import { ComponentBaseProps } from '@/types/global.types';

import styles from './error-message.module.scss';

interface ErrorMessageProps extends ComponentBaseProps {
  hasError?: boolean;
  errorMessage?: string;
}

export const ErrorMessage = ({
  hasError = true,
  errorMessage = '',
  className,
  dataTestId,
}: ErrorMessageProps) => {
  if (!hasError) return null;

  return (
    <p
      data-testid={dataTestId}
      className={clsx([styles.errorMessage, className])}
    >
      {errorMessage}
    </p>
  );
};
