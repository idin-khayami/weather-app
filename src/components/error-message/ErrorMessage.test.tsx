import { render, screen } from '@testing-library/react';

import { ErrorMessage } from './ErrorMessage';

describe('Component: error-message', () => {
  it('should match the snapshot with default params', () => {
    render(<ErrorMessage dataTestId="testingErrorMessage" />);

    expect(screen.getByTestId('testingErrorMessage')).toMatchSnapshot();
  });

  it('should match the snapshot with specified error message', () => {
    render(
      <ErrorMessage
        dataTestId="testingErrorMessage"
        errorMessage="Oops! there was an error!"
      />,
    );

    expect(screen.getByTestId('testingErrorMessage')).toMatchSnapshot();
  });

  it('should render with error message when there is an error', () => {
    const errorMessageText = 'This is an error message';

    render(<ErrorMessage hasError={true} errorMessage={errorMessageText} />);

    expect(screen.getByText(errorMessageText)).toBeInTheDocument();
  });

  it('should not render when there is no error', () => {
    render(
      <ErrorMessage
        dataTestId="testingErrorMessage"
        hasError={false}
        errorMessage="Some error"
      />,
    );

    expect(screen.queryByTestId('testingErrorMessage')).toBeNull();
  });

  it('should render by default', () => {
    render(<ErrorMessage dataTestId="testingErrorMessage" />);

    const errorMessage = screen.getByTestId('testingErrorMessage');

    expect(errorMessage).toBeInTheDocument();
    expect(errorMessage).toHaveTextContent('');
  });
});
