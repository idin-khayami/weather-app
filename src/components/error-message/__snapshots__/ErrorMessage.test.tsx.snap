// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: error-message should match the snapshot with default params 1`] = `
<p
  class="errorMessage"
  data-testid="testingErrorMessage"
/>
`;

exports[`Component: error-message should match the snapshot with specified error message 1`] = `
<p
  class="errorMessage"
  data-testid="testingErrorMessage"
>
  Oops! there was an error!
</p>
`;
